src/Common/getNumberOfPhysicalCPUCores.cpp

```diff
@@ -9,6 +9,15 @@
 
 unsigned getNumberOfPhysicalCPUCores()
 {
+#if CLICKHOUSE_HARDWARE_INFO_FROM_ENV
+    const char * raw_env_cores = std::getenv("CLICKHOUSE_CPU_CORES");
```

Comment:
Maybe it is better to provide a config directive and use `from_env` modifier in config if you need to override it from env?

i.e.

```xml
<yandex>
    <cpu_cores from_env="CLICKHOUSE_CPU_CORES" />
</yandex>
```

And I think that detecting it automatically should be done in a first place (as already suggested by @amosbird ), this option should be used only if automatic detection does not work correctly.

Reply:
Ok.I will refine it later.